#Copyright 2012 Cumulus Networks, Inc.  All rights reserved.

The kernel limits the number of IGMP (IP multicast) groups that a host can join.
There is a sysctl field, igmp_max_memberships, that both adjusts this as well as
can change away from the default.

Since we're a router with about 64 ports and OSPF, the most popular IGP, uses IP
multicast for hellos, the default number of groups that we can joing has been
increased to 96.

diff --git a/net/ipv4/igmp.c b/net/ipv4/igmp.c
index 75b0860..79c0977 100644
--- a/net/ipv4/igmp.c
+++ b/net/ipv4/igmp.c
@@ -106,7 +106,7 @@
 #include <linux/seq_file.h>
 #endif
 
-#define IP_MAX_MEMBERSHIPS	20
+#define IP_MAX_MEMBERSHIPS	96
 #define IP_MAX_MSF		10
 
 #ifdef CONFIG_IP_MULTICAST
